product-service/tests/handler.test.ts

```diff
@@ -23,3 +24,33 @@ test("getProductById", async () => {
     expect(response.statusCode).toEqual(400);
     expect(typeof response.body).toBe("string");
 });
+
+test("catalogBatchProcess", async () => {
+    const record1 = {
+        get body() {
+            return '{"title":"Watch 10","description":"Watch Description 10","price":"12","count":"1","image_url":"https://source.unsplash.com/daily/?watch","image_title":"Watch Description 10"}'
+        }
+    };
+    const record2 = {
+        get body() {
+            return '{"title":"Watch 12","description":"Watch Description 12","price":"18","count":"0","image_url":"https://source.unsplash.com/daily/?watch","image_title":"Watch Description 12"}'
+        }
+    };
+    const spy1 = jest.spyOn(record1, 'body', 'get');
+    const spy2 = jest.spyOn(record2, 'body', 'get');
+    const event = {
+        Records: [
+            record1,
+            record2
+        ]
+    } as SQSEvent;
+    const context = {} as Context;
+    Client.connect = jest.fn();
+    Client.query = jest.fn();
+    Client.end = jest.fn();
+
+    await handler.catalogBatchProcess(event, context);
```

Comment:
You invoke a real SNS publish method during tests. Also it seems that the same situation is with db calls.
You haven't covered by tests anything inside your function

Reply:
Agree

Reply:
Thank you for this valuable feedback. I've mocked DB calls. SNS mock is next in line.